sklearn/datasets/_arff_parser.py

```diff
@@ -265,4 +299,173 @@ def _liac_arff_parser(
         elif y.shape[1] == 0:
             y = None
 
-    return X, y, frame, nominal_attributes
+    if output_arrays_type == "pandas":
+        return X, y, frame, None
+    return X, y, None, nominal_attributes
+
+
+def _pandas_arff_parser(
+    gzip_file,
+    output_arrays_type,
+    columns_info_openml,
+    feature_names_to_select,
+    target_names_to_select,
+):
+    """ARFF parser using `pandas.read_csv`.
+
+    The metadata being fetch directly to OpenML, one can skip the metadata from
+    the ARFF file and read the data as a CSV file.
+
+    Parameters
+    ----------
+    gzip_file : GzipFile instance
+        The file compressed to be read.
+
+    output_array_type : {"numpy", "sparse", "pandas"}
+        The type of the arrays that will be returned. The possibilities ara:
+
+        - `"numpy"`: both `X` and `y` will be NumPy arrays;
+        - `"sparse"`: `X` will be sparse matrix and `y` will be a NumPy array;
```

Comment:
It's not possible to load sparse ARFF datasets with pandas, right?

Reply:
Right. We are currently erroring before even reading the data. You can use sparse dtype but I am not sure that it is straightforward to implement on the top of the `read_csv`. There is a little interest if we cannot do it while opening the CSV.